pkg/envoy/lds/builder_test.go

```diff
@@ -0,0 +1,336 @@
+package lds
+
+import (
+	"testing"
+
+	xds_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	xds_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	xds_tcp_proxy "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/tcp_proxy/v3"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/rds/route"
+	"github.com/openservicemesh/osm/pkg/identity"
+	"github.com/openservicemesh/osm/pkg/service"
+	"github.com/openservicemesh/osm/pkg/tests"
+	"github.com/openservicemesh/osm/pkg/trafficpolicy"
+)
+
+func TestHTTPConnManagerBuilder(t *testing.T) {
+	contains := func(filters []*xds_hcm.HttpFilter, filterName string) bool {
+		for _, f := range filters {
+			if f.Name == filterName {
+				return true
+			}
+		}
+		return false
+	}
+
+	testCases := []struct {
+		name       string
+		buildFunc  func(b *httpConnManagerBuilder)
+		assertFunc func(*assert.Assertions, *xds_hcm.HttpConnectionManager)
```

Comment:
and then the assert could be on the actual bytes/yaml definition of the output from the builder. That said it may hurt readability to do so

Reply:
I feel asserting on the fields is easier as we may not want to compare  the whole object

Reply:
Agree with Shashank here; this feels like the more readable option